### PR TITLE
Mac cross build

### PR DESCRIPTION
**Pull request by bfolie**
Add advice for what to do when FindJNI does not work. Also remove support for 2.12

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import Dependencies._
 
-ThisBuild / scalaVersion       := "2.12.10"
-ThisBuild / crossScalaVersions := List("2.12.10", "2.13.5")
+ThisBuild / scalaVersion       := "2.13.5"
+ThisBuild / crossScalaVersions := List("2.13.5")
```
Comment (bfolie) [inline build.sbt]: I don't think that we're using Scala 2.12 any longer

File: src/native/CMakeLists.txt [hunk 2]
```diff
@@ -11,7 +11,12 @@ cmake_minimum_required(VERSION 3.0.0)
 project (ecos)
 set(PROJECT_VERSION_MAJOR 0)
 set(PROJECT_VERSION_MINOR 0)
-set(PROJECT_VERSION_PATCH 1)
+set(PROJECT_VERSION_PATCH 8)
```
Comment (bfolie) [inline src/native/CMakeLists.txt]: Does this version do anything? Should it be bumped along with the version in sbt? Should it be bumped when this file is changed?

File: build.sbt (hunk 1 above)
Comment (sfriedowitz) [reply build.sbt]: We are not. You can get rid of the `crossScalaVersions` key entirely, I think.

File: src/native/CMakeLists.txt [hunk 3]
```diff
@@ -11,7 +11,12 @@ cmake_minimum_required(VERSION 3.0.0)
...
+
+set(JAVA_HOME "$ENV{JAVA_HOME}")
+set(JAVA_INCLUDE_PATH "${JAVA_HOME}/include")
+set(JAVA_INCLUDE_PATH2 "${JAVA_HOME}/include/darwin")
+set(JAVA_AWT_INCLUDE_PATH "${JAVA_HOME}/include")
```
Comment (bfolie) [inline src/native/CMakeLists.txt]: https://gitlab.kitware.com/cmake/cmake/-/issues/23364

File: src/native/CMakeLists.txt (hunk 2 above)
Comment (sfriedowitz) [reply src/native/CMakeLists.txt]: I have no idea. I have never touched it in the previous bumps.

**Comment by sfriedowitz** [comment]
One other thing that I forgot to do in my PR. This repo has a dependency `apache-slf4j-log4j`. I don't think we use logging at all inside this repo, so we can probably remove that?

**Comment by bfolie** [comment]
The build failed when it was unable to find `jni_md.h`. So clearly these changes are not harmless. I will include them in the README as advice if FIND_JNI isn't working on your local machine, but leave `CMakeLists.txt` unchanged

File: build.sbt [hunk 4]
```diff
@@ -1,7 +1,6 @@
 import Dependencies._
 
-ThisBuild / scalaVersion       := "2.12.10"
-ThisBuild / crossScalaVersions := List("2.12.10", "2.13.5")
+ThisBuild / scalaVersion       := "2.13.5"
 ThisBuild / version            := "0.0.8"
```
Comment (sfriedowitz) [inline build.sbt]: Need a version bump

Comment (bfolie) [reply build.sbt]: Seems silly because the artifacts aren't changing. But we don't pay by the version 🤷

**Review by sfriedowitz — APPROVED** [review]
LGTM

**Pull request merged**